### PR TITLE
Data cleaning function created

### PR DESCRIPTION
**Pull request by g-christodoulou**

### CODE REVIEW

File: Data Cleaning.Rmd [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  # remove rows where FLAG_OWN_CAR = "Y" and OWN_CAR_AGE is NA
+  data <- data[!(data$FLAG_OWN_CAR == "Y" & is.na(data$OWN_CAR_AGE)), ]
+
+  # add 1 year to all non-NA values of OWN_CAR_AGE
+  data$OWN_CAR_AGE <- ifelse(!is.na(data$OWN_CAR_AGE), data$OWN_CAR_AGE + 1, data$OWN_CAR_AGE)
+
+  # replace remaining NAs in OWN_CAR_AGE with 0
+  data$OWN_CAR_AGE[is.na(data$OWN_CAR_AGE)] <- 0
+
+  # replace NAs in EXT columns with the mean of the column
```
Comment (adambushman) [inline Data Cleaning.Rmd]: We were going to do two things:
1. Add a binary/boolean column for each "EXT_" that indicated if it was an imputed value
2. Looking at the distributions, we may want to switch to "median"; "2" and "3" are skewed to the right.
ext_source_2
ext_source_3

File: Data Cleaning.Rmd [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```{r}
+knitr::opts_chunk$set(warning = FALSE, message = FALSE)
+```
+
+```{r}
+library(tidyverse)
+library(dbplyr)
+
+
+application_train <- read.csv("data/application_train.csv")
```
Comment (adambushman) [inline Data Cleaning.Rmd]: These take a while to load. You could use `data.table::fread` instead. A lot faster. Here's an example: 

```
application_train <- data.table::fread("data/application_train.csv")
application_test <- data.table::fread("data/application_test.csv")
```

File: Data Cleaning.Rmd [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  return(data)
+}
+
+application_train_clean <- application_cleaning(application_train)
+
+application_test_clean <- application_cleaning(application_test)
+
+application_train_clean %>% summary()
+```
```
Comment (adambushman) [inline Data Cleaning.Rmd:123]: We should add to this script a write to a CSV. That way we only have to run this once each and then for any modeling we just ready from the new, "cleaned" file. We can use `fwrite`, which is really fast, like so:

```
data.table::fwrite(
  application_train_clean, 
  file = "data/application_train_clean.csv", 
  row.names = FALSE
)
```

File: Data Cleaning.Rmd [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  data[data == ""] <- NA
+
+  # impute NAME_TYPE_SUITE with "Unaccompanied"
+  data$NAME_TYPE_SUITE[is.na(data$NAME_TYPE_SUITE)] <- "Unaccompanied"
+
+  # impute OCCUPATION_TYPE with XNA
+  data$OCCUPATION_TYPE[is.na(data$OCCUPATION_TYPE)] <- "XNA"
+
+  # convert all character columns to factors
+  data[sapply(data, is.character)] <- lapply(data[sapply(data, is.character)], factor)
```
Comment (adambushman) [inline Data Cleaning.Rmd:72]: One thing we miss here is `TARGET`; this is really a boolean factor, right?

Comment (adambushman) [reply Data Cleaning.Rmd:72]: Also, there's a few "flag" values that are booleans, like `FLAG_MOBIL`, `FLAG_EMAIL`, etc. I think those we want to trun to a `<logical>` data type

Comment (adambushman) [reply Data Cleaning.Rmd:72]: I just ran a script looking for any column in `application_train_clean` that has 2 unique values. These are the columns. All these could be booleans.
columns


The interesting ones are some flags use [0,1] (`LIVE_CITY_NOT_WORK_CITY`, ...) while others use [Y,N] (`FLAG_OWN_CAR`, `FLAG_OWN_REALTY`, ...). This should probably be standardized across all boolean-esque columns.

**Review by adambushman — CHANGES_REQUESTED** [review]
Georgia, this is great! Really well done. Your script hits on almost everything we wanted to do. Here's what I saw:

- Removed only 0.1% of rows in `application_train` 
- Reduced the number of columns in `application_train` by 55%

That's incredible work.

There's a few suggestions I've made and changes I've requested. Shouldn't take much time and then I think we've got it.

**Comment by g-christodoulou** [comment]
Adding the comments here as well for the changes I made!
- Removed 4 "XNA" values from gender so I could turn this to a boolean to stay consistent with the 2 unique values logic
- Changed "target" column name to "default" for easier interpretation
- Changed code_gender to "gender_male" since I changed this to boolean
- Changed contract_type to "cash_loan" since I changed this to boolean
- Added columns for the imputed ext source variables and imputed 2 and 3 with the median instead of mean

**Review by adambushman — APPROVED** [review]
This looks great, Georgia! Really well done.

**Pull request merged**